libs/api_common/device_helper.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include <map>
+#include <stdexcept>
+#include "ebpf_api.h"
+#include "ebpf_bind_program_data.h"
+#include "ebpf_platform.h"
+#include "ebpf_program_types.h"
+#include "ebpf_protocol.h"
+#include "ebpf_result.h"
+#include "ebpf_xdp_program_data.h"
+#include "platform.h"
+#undef VOID
+#include "platform.hpp"
+// #include "device_helper.hpp"
+
+ebpf_handle_t device_handle = INVALID_HANDLE_VALUE;
+
+uint32_t
```

Comment:
should this return ebpf_result_t?

Reply:
initialize_device_handle() is called by ebpf_service_initialize() and ebpf_api_initiate() both of which return uint32_t.

Reply:
Also, as with invoke_ioctl(), initialize_device_handle() returns error code from CreateFile(), so I think it is better to directly return that error code from initialize_device_handle()?

Reply:
Is initialize_device_handle called from multiple places that would need different conversion?  If not, then use ebpf_result_t... push it down as far as possible.

Reply:
Fixed.